### PR TITLE
introduce perf_event programs

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR adds support for the `perf_event` program type. This program type can be used to trigger eBPF programs on software and hardware perf events.

### Motivation

This feature will be used in future eBPF powered use cases in the Datadog Agent.

### Describe how to test your changes

A test program was added in the examples. It should print the active pid every second for all the CPU cores (returned by `runtime.NumCPU()`, so it might not be the real CPU count).

### CODE REVIEW

File: probe.go [hunk 1]
```diff
@@ -960,3 +999,52 @@ func (p *Probe) detachLSM() error {
...
+	if p.PerfEventCPUCount == 0 {
+		p.PerfEventCPUCount = runtime.NumCPU()
+	}
+
+	pid := p.PerfEventPID
+	if pid == 0 {
+		pid = -1
+	}
+
+	for cpu := 0; cpu < p.PerfEventCPUCount; cpu++ {
```
Comment (r1viollet) [inline probe.go:1029]: If you have a partial attachement, ie only part of the perf events are successful and you have a failure in the middle of the loop, should you free those that were successful ?

**Review by r1viollet — APPROVED** [review]
Awesome change thanks. I only left a minor comment on error management. I'm not a competent Go dev, so you might want another review. 
Also the examples look great. Should we consider running some in CI ?

File: probe.go (hunk 1 above)
Comment (Gui774ume) [reply probe.go:1029]: Hey Erwan,
Thanks for the review ! So yes you're 100% right, we need to cleanup successful attempts if the last one failed. The good news is that the `Probe.attach()` method already handles it for us: if `Probe.attachPerfEvent()` returns an error, then `Probe.attach()` will eventually call `_ = p.stop(false)` which will call `Probe.detachPerfEvent()` ... which will free the successful attempts 😃

**Comment by Gui774ume** [comment]
We've relied on the Datadog Agent CI so far ... we should build one for this repo for sure, but haven't had time yet 😭

**Review by safchain — APPROVED** [review]

File: probe.go (hunk 1 above)
Comment (r1viollet) [reply probe.go:1029]: Thanks for pointing that out !

**Pull request merged**